### PR TITLE
Upgrade Code to Python 3

### PR DESCRIPTION
**Pull request by omuzzyEBSCO**
Updating to Python 3 - do not merge (work in progress).

### CODE REVIEW

File: ebscopy/ebscopy.py [hunk 1]
```diff
@@ -50,13 +50,14 @@ def _parse_highlight(text):
...
 	"""
+	output = {} #this is OM addition
 	output["orig"]							= text
 	output["clean"]							= ""
 	output["start_pos"]						= 0
 	output["end_pos"]						= 0
 
 	start_tag								= "&lt;highlight&gt;"
-	end_tag									= "&lt;\/highlight&gt;"
+	end_tag									= "&lt;/highlight&gt;"
```
Comment (jjensen-ebsco) [inline ebscopy/ebscopy.py:60]: Is the linting suggesting this?

File: ebscopy/ebscopy.py [hunk 2]
```diff
@@ -240,7 +242,7 @@ def _change_eds_date_limiter_to_api(l):
 	# 1-19780515	(bad start)
 	# 19780515-1	(bad end)
 	#
-	if l["Id"] != "DT1" or not re.search("\A\s*(?=-?\d{8})(?:\d{8})?-(?:\d{8})?\Z", l["Values"][0]):
+	if l["Id"] != "DT1" or not re.search("\\A\\s*(?=-?\\d{8})(?:\\d{8})?-(?:\\d{8})?\\Z", l["Values"][0]):
```
Comment (jjensen-ebsco) [inline ebscopy/ebscopy.py:245]: Is this a fix, a change due to version, or just a linter preference?

File: ebscopy/ebscopy.py (hunk 1 above)
Comment (omuzzyEBSCO) [reply ebscopy/ebscopy.py:60]: Yes - with the backslash as it was, pylint gave this error: Anomalous backslash in string: '\/'. String constant might be missing an r prefix.

File: ebscopy/ebscopy.py (hunk 2 above)
Comment (omuzzyEBSCO) [reply ebscopy/ebscopy.py:245]: Linting picked this up - adding the additional backslash cleared the "Anomalous backslash in string: '\A'. String constant might be missing an r prefix." warning.

File: ebscopy/ebscopy.py [hunk 3]
```diff
@@ -376,15 +375,15 @@ def request(self, method, data, session_token=None, attempt=0):
 
 		try:
 			headers['x-authenticationToken']	= self.auth_token
-		except:
-			if method is not "UIDAuth":
-				raise ValueError("Missing Authentication Token!")
+		except ValueError as e:
+			if method != "UIDAuth":
```
Comment (jjensen-ebsco) [inline ebscopy/ebscopy.py:379]: Is this correct?
https://realpython.com/python-is-identity-vs-equality/

File: ebscopy/ebscopy.py (hunk 2 above)
Comment (omuzzyEBSCO) [reply ebscopy/ebscopy.py:245]: I think I need to review this concept - I think the additional backslash is escaping the character entirely.

File: ebscopy/ebscopy.py (hunk 3 above)
Comment (omuzzyEBSCO) [reply ebscopy/ebscopy.py:379]: No, it appears not :)

File: ebscopy/ebscopy.py (hunk 2 above)
Comment (jjensen-ebsco) [reply ebscopy/ebscopy.py:245]: Yeah, I don't trust the linter on complex regular expressions. We might have to simplify the regular expression or definite it more cleanly.

File: ebscopy/ebscopy.py (hunk 1 above)
Comment (jjensen-ebsco) [reply ebscopy/ebscopy.py:60]: Might have to define this more intelligently. I assume I did this so I could use it multiple times later on.

**Pull request merged**